### ISSUE TITLE
meta: Use project-level account on PyPI

### ISSUE THREAD

**Issue by dgw** [sopel-irc/sopel#1921]
Enough people are getting involved in this project now that I'm not sure anymore about being listed as the author on PyPI. It's really not just me, even ignoring the many times project ownership changed (Phenny->Jenni->Willie).

So I'd like to update the authorship metadata on PyPI, effective for Sopel 8.0. We should say that "Sopel IRC Contributors" (or a similar name) is the author, and list currently active maintainers as maintainers instead.

PyPI has no concept of "team accounts", which is unfortunate. We'll have to make sure a few people always have (secure) access to the project account's credentials, so theoretically there will always be at least one person who can add more maintainers when (not if) someone just suddenly disappears one day. (I fully acknowledge that that _could_ be me—bus factor and all that.)

Other things to think about includes how to handle releases. It's kind of neat to see who released each Sopel version, and if we can store multiple PyPI tokens to use depending on who pushed the release tag, that would be neat. But the simplest option is obviously to continue using one account—the "primary" maintainer's, or the project's—to deploy releases.\*

----

\* — No idea if mapping the tagging user to release credentials is possible; it's just a cool idea. Moving off of Travis CI would affect this, too—we've tossed around the idea of switching to GitHub Actions.

**Comment by Exirel** [comment]
In #2328 there is an optional change that can be brought to have 2 types of information:

* a list of **authors** (and you could be one)
* a list of **maintainers** (and other people could be in that list)

I don't think it's important to know who released a version. Currently, you need to have the PyPI token to do so, or to have the right to tag/publish a release on the Github repository anyway, and I think you are the only one that can do so?

In any case, I'll be happy to be considered a maintainer, with all the responsibilities that come with it. Just don't ask me to write the next release note without grammar issues. :grin:

**Comment by dgw** [comment]
Yeah, OK. De-milestoning because we're _so close_ to being able to do this in a nice way, but waiting on PyPI.

![image](https://github.com/sopel-irc/sopel/assets/164140/31941c9a-8ac5-43cc-932c-dfcc134e1684)

(I definitely thought the request would be processed in less than the month it's been sitting so far.)

**Comment by dgw** [comment]
Approaching 19 months, no change (except to the page styles).

![Image](https://github.com/user-attachments/assets/37553791-ab22-4ce8-a66e-7b0422900462)

**Comment by dgw** [comment]
There is _some_ news on this, but no progress yet. 70 community organizations have been admitted to the private beta so far, and there are currently 9,182 applications pending including ours.

No promises made yet as to a timeline for opening it up further. https://discuss.python.org/t/state-of-pypi-organizations/33764/25

**Comment by dgw** [comment]
Hot off the ~~presses~~ email inbox ~25 minutes ago:

> ### [PyPI] Your new organization named 'sopel' has been approved and created
> 
> Your new organization named 'sopel' has been approved and created.
> 
> You are the owner of the new organization and can log in to manage organization projects, collaborators, teams, or settings.
> 
> Welcome to PyPI Orgs!

Finally I'll be able to get going on this, maybe over the weekend (which, conveniently, is tomorrow). 💪
